### DISCUSSION TITLE
IBM i 7.3

### DISCUSSION THREAD

**Discussion by ibmiiste** [codefori/vscode-ibmi#1455] [Q&A]
Hi, 

I will want to use Code for IBM i, but we are in 7.3 version, how can we use Code for IBM i with prior versions?

Regards, 

Olivier.

**Comment by worksofliam** [comment]
It might work for 7.2 but we only officially support 7.3 and above.

**Comment by ibmiiste** [comment]
Ok, 

Thank you for your reply.

However, not all function are available, when I want to work locally (I would like using Git and I think it is by this way) and then deploy, I got this problem :

CRTSQLRPGI OBJ(L000000411/example) SRCSTMF('/home/DRUBIGNY/builds/Test/source/example.sqlrpgle') OBJTYPE(*MODULE) CLOSQLCSR(*ENDMOD) OPTION(*EVENTF) DBGVIEW(*SOURCE) TGTRLS(*CURRENT) CVTCCSID(*JOB)
CPE3490:  Conversion error.

And if I change by 
RTSQLRPGI OBJ(L000000411/example) SRCSTMF('/home/DRUBIGNY/builds/Test/source/example.sqlrpgle') OBJTYPE(*MODULE) CLOSQLCSR(*ENDMOD) OPTION(*EVENTF) DBGVIEW(*SOURCE) TGTRLS(*CURRENT)
I got :
CPE3490:  Conversion error.

The CVTCCSID parameter is not supported by 7.3 yet.

So what can I do? 

Regards, 

Olivier.

**Reply by worksofliam** [reply]
You could tag the stream file with a valid CCSID instead of using CVTCCSID. I have used 1252 in the past.

**Comment by chrjorgensen** [comment]
@ibmiiste If you have your source code in IFS in CCSID 1208 (UTF-8), you're not using the correct compile command - you have to use `TGTCCSID(*JOB)` (or a valid CCSID like 37) for the `CRTBNDRPG` and `CRTRPGMOD` commands. And when you compile SQLRPGLE source, you must specify parameters `RPGPPOPT(*LVL1)` or `RPGPPOPT(*LVL2)` and `COMPILEOPT('TGTCCSID(*JOB)')`.
This has been available for IBM i 7.1, 7.2 and 7.3 since 2016.

For more information, see here: https://www.ibm.com/support/pages/node/1108071

The `CVTCCSID` parameter was added to the `CRTSQLRPGI` command in IBM i 7.5 and will probably not be backported to 7.4 or 7.3.

**Comment by ibmiiste** [comment]
Hi, 
the PTFs needed (SI62605 and SE72036) are not installed on our system :-( . So I have to put pressure on my hierarchy :-).
Regards,

**Reply by chrjorgensen** [reply]
@ibmiiste Please put pressure for updating the system!! Not having seven years old fixes is a huge risk and scares the H... out of me!

Ideally the system should be upgraded to 7.5, since 7.3 goes out of support after September.

But at least the latest PTF groups should be applied. 🙏

Hope you can convince your upper management. Not installing fixes is NOT best practice. 😔
